src/us/kbase/typedobj/db/MongoTypeStorage.java

```diff
@@ -54,36 +54,45 @@ public MongoTypeStorage(DB db) {
 		ensureIndeces();
 	}
 	
+	private static final DBObject UNIQ = new BasicDBObject("unique", true);
+	
 	private void ensureIndeces() {
-		MongoCollection reqs = jdb.getCollection(TABLE_MODULE_REQUEST);
-		reqs.ensureIndex("{moduleName:1,ownerUserId:1}", "{unique:true}");
-		reqs.ensureIndex("{ownerUserId:1}", "{unique:false}");
-		MongoCollection vers = jdb.getCollection(TABLE_MODULE_VERSION);
-		vers.ensureIndex("{moduleName:1}", "{unique:true}");
-		MongoCollection owns = jdb.getCollection(TABLE_MODULE_OWNER);
-		owns.ensureIndex("{moduleName:1,ownerUserId:1}", "{unique:true}");
-		owns.ensureIndex("{ownerUserId:1}", "{unique:false}");
-		MongoCollection infos = jdb.getCollection(TABLE_MODULE_INFO_HISTORY);
-		infos.ensureIndex("{moduleName:1,versionTime:1}", "{unique:true}");
-		MongoCollection specs = jdb.getCollection(TABLE_MODULE_SPEC_HISTORY);
-		specs.ensureIndex("{moduleName:1,versionTime:1}", "{unique:true}");
-		MongoCollection tschs = jdb.getCollection(TABLE_MODULE_TYPE_SCHEMA);
-		tschs.ensureIndex("{moduleName:1,typeName:1,version:1}", "{unique:true}");
-		tschs.ensureIndex("{moduleName:1,moduleVersion:1}", "{unique:false}");
-		MongoCollection tprs = jdb.getCollection(TABLE_MODULE_TYPE_PARSE);
-		tprs.ensureIndex("{moduleName:1,typeName:1,version:1}", "{unique:true}");
-		tprs.ensureIndex("{moduleName:1,moduleVersion:1}", "{unique:false}");
-		MongoCollection fprs = jdb.getCollection(TABLE_MODULE_FUNC_PARSE);
-		fprs.ensureIndex("{moduleName:1,funcName:1,version:1}", "{unique:true}");
-		fprs.ensureIndex("{moduleName:1,moduleVersion:1}", "{unique:false}");
-		MongoCollection frefs = jdb.getCollection(TABLE_FUNC_REFS);
-		frefs.ensureIndex("{depModule:1,depName:1,depVersion:1}", "{unique:false}");
-		frefs.ensureIndex("{refModule:1,refName:1,refVersion:1}", "{unique:false}");
-		frefs.ensureIndex("{depModule:1,depModuleVersion:1}", "{unique:false}");
-		MongoCollection trefs = jdb.getCollection(TABLE_TYPE_REFS);
-		trefs.ensureIndex("{depModule:1,depName:1,depVersion:1}", "{unique:false}");
-		trefs.ensureIndex("{refModule:1,refName:1,refVersion:1}", "{unique:false}");
-		trefs.ensureIndex("{depModule:1,depModuleVersion:1}", "{unique:false}");
+		final DBCollection reqs = db.getCollection(TABLE_MODULE_REQUEST);
+		reqs.createIndex(new BasicDBObject("moduleName", 1).append("ownerUserId", 1), UNIQ);
```

Comment:
Do these raise any error if the index already exists?

Reply:
Nope

Reply:
Oh, I see why you were thinking that. They deprecated `ensureIndex` and made `createIndex` the replacement.